### ISSUE TITLE
About training part of the code

### ISSUE THREAD

**Issue by li1608335419** [OneForAllSama/MTS#1]
Dear author, I did not see any code related to the training section in the document. Could you please provide me with the training code?

**Comment by Xp1307** [comment]
> Dear author, I did not see any code related to the training section in the document. Could you please provide me with the training code?

    num_epochs = 20
    best_acc = 0.0

    for epoch in range(num_epochs):
        print(f"\nEpoch [{epoch + 1}/{num_epochs}]")

        # training mode
        image_encoder.train()
        text_model.train()
        multi_trans_model.train()

        total_loss = 0
        total_correct = 0
        total_samples = 0

        for text, image, label in tqdm(train_loader, desc="Training"):
            optimizer.zero_grad()

            image = image.to(device)
            label = label.to(device)
            
            # text input
            inputs = text_tokenizer(text, return_tensors='pt', padding=True, truncation=True).to(device)
            t_outs = text_model(**inputs)
            
            # image input
            i_outs = image_encoder(image)
            
            # multi trans ouput
            final_out, _, _, _, _, _, _ = multi_trans_model(t_outs.pooler_output, i_outs, return_useful=False)

            loss = loss_fn(final_out, label)
            loss.backward()
            optimizer.step()

            total_loss += loss.item()
            total_correct += (final_out.argmax(1) == label).sum().item()
            total_samples += len(label)

        avg_loss = total_loss / total_samples
        acc = total_correct / total_samples
        print(f"Train Loss: {avg_loss:.4f}, Train Acc: {acc:.4f}")

        # test, but the code doesn't include the valid dataset
        test_acc, test_loss, report, _ = test(test_loader, plot_tsne=False)

        # save the best test acc model
        if test_acc > best_acc:
            best_acc = test_acc
            torch.save(image_encoder, "best_tailor_img_encoder.pt")
            torch.save(text_model, "best_tailor_text_encoder.pt")
            torch.save(multi_trans_model, "best_tailor_multi_tailor_model.pt")
            print(f"*** Best Model Saved! Val Acc: {test_acc:.4f}")

    print(f"\nTraining Complete. Best Val Acc: {best_acc:.4f}")


I guess this is probably the training part.
